ngraph/core/src/op/util/broadcast_base.cpp

```diff
@@ -159,7 +160,8 @@ void op::util::BroadcastBase::validate_target_shape_none(const PartialShape& arg
         if (arg_shape.rank().get_length() > 0)
         {
             NODE_VALIDATION_CHECK(this,
-                                  target_shape[axes_mapping_val[i]].same_scheme(arg_shape[i]),
+                                  target_shape[axes_mapping_val[i]].is_dynamic() ||
+                                      target_shape[axes_mapping_val[i]].same_scheme(arg_shape[i]),
```

Comment:
As above - please try with target_shape[axes_mapping_val[0]].compatible(arg_shape[i]).
Also it'd be good if you could add some type_prop tests for Broadcast that cover those cases.